### PR TITLE
QEP #5: stricter unit tests

### PR DESCRIPTION
**Pull request by nyalldawson**
This proposal is aimed at providing more stable releases and preventing regressions in QGIS releases by expanding the coverage of unit tests and adopting a hard line approach to compulsory unit tests.

### CODE REVIEW

File: QEP-5-Unit-tests.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+that they will not suffer regressions in future QGIS releases. Commits which are not accompanied by unit tests will not be accepted
+into master. Code changes for GUI and app are immune from this requirement due to the complexity in writing interactive
+tests suites. 
+
+Exemption for specific commits may be granted, but only after discussion on the QGIS developer list.
+
+Generally, unit tests in either c++ or python are acceptable (or both). However, methods which involve transfer of ownership
+of objects must include python tests to ensure that this transfer is handled correctly.
+
+To enforece this requirement, commits which modify core and which are not accompanied by unit tests (or have been granted
```
Comment (strk) [inline QEP-5-Unit-tests.rst]: typo

**Comment by strk** [comment]
How can this QEP be moved on ?

**Comment by nyalldawson** [comment]
Superseded by #23 - I don't think we're quite ready for this one yet.

**Comment by m-kuhn** [comment]
No need to supersede, we can just leave it open until we are ready :-)

**Comment by mhugo** [comment]
Sorry, maybe I missed something. Why aren't we not ready yet ?

**Comment by m-kuhn** [comment]
Because of the lack of a broad set of tests to build upon, a good set of rules of when this applies and acceptance in general IMHO.

**Comment by NathanW2** [comment]
QEPs done via issues now.

New ticket at #42

Sorry for the inconvenience.

Email out soon on why.

**Pull request closed without merging**